docs/api/standard_library_types.md

```diff
@@ -43,3 +47,89 @@ except ValidationError as e:
       Input should be a valid boolean [type=bool_type, input_value=[], input_type=list]
     """
 ```
+
+## Number Types
+
+## Datetime Types
+
+Pydantic supports the following [datetime](https://docs.python.org/library/datetime.html#available-types)
+types:
+
+### `datetime.datetime`
+* `datetime` fields will accept values of type:
+
+    * `datetime`; an existing `datetime` object
+    * `int` or `float`; assumed as Unix time, i.e. seconds (if >= `-2e10` and <= `2e10`) or milliseconds
+      (if < `-2e10`or > `2e10`) since 1 January 1970
+    * `str`; the following formats are accepted:
+        * `YYYY-MM-DD[T]HH:MM[:SS[.ffffff]][Z or [±]HH[:]MM]`
+        * `int` or `float` as a string (assumed as Unix time)
+
+### `datetime.date`
+* `date` fields will accept values of type:
+
+    * `date`; an existing `date` object
+    * `int` or `float`; handled the same as described for `datetime` above
+    * `str`; the following formats are accepted:
+        * `YYYY-MM-DD`
+        * `int` or `float` as a string (assumed as Unix time)
+
+### `datetime.time`
+* `time` fields will accept values of type:
+
+    * `time`; an existing `time` object
+    * `str`; the following formats are accepted:
+        * `HH:MM[:SS[.ffffff]][Z or [±]HH[:]MM]`
+
+### `datetime.timdelta`
+* `timedelta` fields will accept values of type:
+
+    * `timedelta`; an existing `timedelta` object
+    * `int` or `float`; assumed to be seconds
+    * `str`; the following formats are accepted:
+        * `[-][DD ][HH:MM]SS[.ffffff]`
+        * `[±]P[DD]DT[HH]H[MM]M[SS]S` ([ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format for timedelta)
+
+```py
+from datetime import date, datetime, time, timedelta
+
+from pydantic import BaseModel
+
+
+class Model(BaseModel):
+    d: date = None
+    dt: datetime = None
+    t: time = None
+    td: timedelta = None
+
+
+m = Model(
+    d=1679616000.0,
+    dt='2032-04-23T10:20:30.400+02:30',
+    t=time(4, 8, 16),
+    td='P3DT12H30M5S',
+)
+
+print(m.model_dump())
+"""
+{'d': datetime.date(2023, 3, 24), 'dt': datetime.datetime(2032, 4, 23, 10, 20, 30, 400000, tzinfo=TzInfo(+02:30)), 't': datetime.time(4, 8, 16), 'td': datetime.timedelta(days=3, seconds=45005)}
+"""
+```
```

Comment:
Instead of a single example at the end with all the types of the section, maybe we can have smaller ones for each type here?

Reply:
Sure thing. I like this new pattern better, as it encourages us to have an example for every standard type 😄.